File: scripts/scrape-bravo.js
import requests
from bs4 import BeautifulSoup
import json
from datetime import datetime

def scrape_bravo_menu():
    url = "https://bravobrewingcompany.com/"
    response = requests.get(url)
    soup = BeautifulSoup(response.text, 'html.parser')
    
    menu_sections = []
    
    # Target each menu section (On Tap, Guest Taps)
    sections = soup.find_all('div', class_='section')
    
    for section in sections:
        header = section.find('h3', class_='section-name')
        if not header:
            continue
            
        section_data = {
            "header": header.get_text(strip=True),
            "items": []
        }
        
        # Target each menu item within the section
        items = section.find_all('div', class_='menu-item')
        for item in items:
            beer_data = {
                "name": item.find('span', id=lambda x: x and 'description' not in x).get_text(strip=True) if item.find('span', id=lambda x: x and 'description' not in x) else "N/A",
                "style": item.find('span', class_='item-category').get_text(strip=True) if item.find('span', class_='item-category') else "",
                "abv": item.find('span', class_='item-abv').get_text(strip=True) if item.find('span', class_='item-abv') else "",
                "description": item.find('p', class_='show-less').get_text(strip=True) if item.find('p', class_='show-less') else ""
            }
            section_data["items"].append(beer_data)
            
        menu_sections.append(section_data)
    
    # Final structure for your JSON file
    final_output = {
        "Beer on tap": menu_sections,
        "lastUpdated": datetime.now().strftime("%m/%d/%Y, %I:%M:%S %p")
    }
    
    return json.dumps(final_output, indent=2)

# Run the scrape and save to file
with open('public/data/bravo_menu.json', 'w') as f:
    f.write(scrape_bravo_menu())

File: scripts/scrape-bravo.py
from playwright.sync_api import sync_playwright
from bs4 import BeautifulSoup
import json
import os

def scrape_menu():
    url = "https://bravobrewingcompany.com/" 
    
    # 1. Use Playwright to launch a browser
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.goto(url)
        
        # Wait for the content to load
        page.wait_for_selector(".menu-item")
        
        # Get the full page content
        html = page.content()
        browser.close()
    
    # 2. Parse the HTML content
    soup = BeautifulSoup(html, 'html.parser')
    
    # 3. Find all beer containers
    items = soup.select('.menu-item')
    print(f"Total menu items found: {len(items)}")
    
    menu_items = []
    
    # 4. Loop through items
    for item in items:
        name_el = item.select_one('.item-name span')
        style_el = item.select_one('.item-category')
        abv_el = item.select_one('.item-abv')
        desc_el = item.select_one('.item-description p')

        name = name_el.get_text(strip=True) if name_el else "N/A"
        style = style_el.get_text(strip=True) if style_el else "N/A"
        abv = abv_el.get_text(strip=True) if abv_el else "N/A"
        description = desc_el.get_text(strip=True) if desc_el else ""

        menu_items.append({
            "name": name,
            "style": style,
            "abv": abv,
            "description": description
        })

    # 5. Save to your app's public data folder
    output_dir = os.path.join('..', 'public', 'data')
    os.makedirs(output_dir, exist_ok=True)
    
    output_path = os.path.join(output_dir, 'bravo_menu.json')
    
    with open(output_path, 'w') as f:
        json.dump({"On Tap": menu_items, "lastUpdated": "6/26/2026"}, f, indent=2)
    
    print(f"Menu saved successfully to {output_path}")

# This must be at the very bottom, aligned to the far left
if __name__ == "__main__":
    scrape_menu()